Upgrade downgrade test clean up

eloycoto opened the pull request:
Hi, 

Nightly is failing in the downgrade+Upgrade test because the changes that has been made in the last days. This PR has two commits: 

- One fix some asserts failures that are not correct and mark the test as skipped. 
- Fix the issues on Upgrade from v1.0, if cannot be installed will skip the test. This commit has a dependency on #5507 PR. 

This should fix the Nigthly builds from 320-327. 

Regards.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cilium/cilium/5508)
<!-- Reviewable:end -->

eloycoto committed:
Test: Cleanup asserts on Upgrade/Downgrade test

Fix some asserts messages, format and error reporting

Signed-off-by: Eloy Coto <eloy.coto@gmail.com>

eloycoto committed:
Test: Fix Nightly test

- Fix issues with kube-dns that are installed before it has been
deleted.
- Fix issues with kubectl instalation that fails on invalid
installations.

Fix Nightly issues from 320 to 327

Signed-off-by: Eloy Coto <eloy.coto@gmail.com>

aanm commented on test/k8sT/Nightly.go:
```diff
@@ -442,11 +439,29 @@ var _ = Describe("NightlyExamples", func() {
 
+					By("Validating that cilium can be installed")
+					canRun, err := helpers.CanRunK8sVersion(version, helpers.GetCurrentK8SEnv())
+
```
Maybe we can move this to `ValidateCiliumUpgrades` to avoid duplicated code.

eloycoto replied:
Is in there, but Validating cilium does not install cilium on that version, so that check need to happens in there.

aanm replied:
I mean from line 447 to line 463

eloycoto replied:
do the installation in the ValidateCiliumUpgrades?

aanm replied:
Yes, and remove the BeforeEach from `test/k8sT/Updates.go`

eloycoto replied:
@aanm done

eloycoto committed:
Test: Upgrade test nightly from 1.1

Update the upgrade table from 1.0 and 1.1 to master  in nightly.

Signed-off-by: Eloy Coto <eloy.coto@gmail.com>

eloycoto committed:
Test: Refactor ValidaCiliumUpgrades

Rename the ValidaCiliumUpgrades test helpers to instal Cilium in the
function itself and avoid duplicated code.

Signed-off-by: Eloy Coto <eloy.coto@gmail.com>
